### ISSUE TITLE
Uncaught exception of tyep std::bad_alloc: std::bad_alloc when running Stellar

### ISSUE THREAD

**Issue by zijuexiansheng** [seqan/seqan#2101]
The following is the uncaught exception when I tried to run Stellar.

```
Constructing index...
/afs/crc.nd.edu/user/s/szhu3/loonlocal/openbiosrc/seqan/include/seqan/basic/basic_exception.h:363 FAILED!  (Uncaught exception of type std::bad_alloc: std::bad_alloc)

stack trace:
  0          [0x4839ae]  seqan::ClassTest::fail() + 0xe
  1          [0x46f298]  stellar()
  2    [0x7f3a8f1880e6]  /opt/crc/gcc/4.9.2/gcc-4.4.7/lib64/libstdc++.so.6(+0x5e0e6)
  3    [0x7f3a8f188131]  /opt/crc/gcc/4.9.2/gcc-4.4.7/lib64/libstdc++.so.6(+0x5e131)
  4    [0x7f3a8f188348]  /opt/crc/gcc/4.9.2/gcc-4.4.7/lib64/libstdc++.so.6(+0x5e348)
  5    [0x7f3a8f188859]  /opt/crc/gcc/4.9.2/gcc-4.4.7/lib64/libstdc++.so.6(+0x5e859)
  6          [0x48e0a5]  seqan::Size<seqan::String<unsigned long, seqan::Alloc<void> > >::Type seqan::_Resize_String<seqan::Tag<seqan::TagExact_> >::resize_<seqan::String<unsigned long, seqan::Alloc<void> > >(seqan::String<unsigned long, seqan::Alloc<void> >&, seqan::Size<seqan::String<unsigned long, seqan::Alloc<void> > >::Type) + 0x55
  7          [0x47d273]  stellar()
  8          [0x56e951]  bool _stellarOnAll<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5_>, seqan::Alloc<void> >, seqan::String<char, seqan::Alloc<void> > >(seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5_>, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, seqan::StringSet<seqan::String<char, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5_>, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, seqan::StringSet<seqan::String<char, seqan::Alloc<void> >, seqan::Owner<seqan::Tag<seqan::Default_> > >&, StellarOptions&) + 0x7c1
  9          [0x56f561]  int mainWithOptions<StellarOptions, seqan::SimpleType<unsigned char, seqan::Dna5_> >(StellarOptions&, seqan::String<seqan::SimpleType<unsigned char, seqan::Dna5_>, seqan::Alloc<void> >) + 0x3e1
 10          [0x4685b7]  main + 0x397
 11    [0x7f3a8e64bb35]  __libc_start_main + 0xf5
 12          [0x46965d]  stellar()

```

The command I used is `stellar ref.fa joined.reads.fa -o joined.gff`

I was just following the readme file to find SVs using gustaf, with a reference genome and Illumina MiSeq pair-end reads. But I got this error. Any idea of the reasons for this uncaught exception?

The issue exists in both master branch (commit a0691b9d05eb1ee56f513a5642f745231d77798b) and `seqan-v2.3.2`

**Comment by smehringer** [comment]
Hi @zijuexiansheng,

we are really sorry for answering so late. I don't know much about Stellar but since the error occurred while constructing the index @cpockrandt might be able to help?
From experience, index construction errors often occur due to memory issues, so make sure you don't exceed your quota limits anywhere.

**Comment by zijuexiansheng** [comment]
@smehringer Thank you very much. I'll try and see if I can solve the issue by myself. At least I have some clues now.

**Comment by smehringer** [comment]
I'm really not sure about this, but in case the index is created on secondary memory in your temp directory, you could try `export TMPDIR=/somewhere/else/with/more/space` to avoid the quota or disk space limitation.

**Comment by cpockrandt** [comment]
@zijuexiansheng This doesn't look like an out-of-secondary-memory error from the index module to me but it's worth a shot. You could also try one of the binaries of stellar (see http://packages.seqan.de/stellar/) and check whether the error occurs there as well or whether it might be a bug in newer SeqAn versions.

**Comment by rrahn** [comment]
What is the state of this issue?

**Comment by zijuexiansheng** [comment]
I tried from Stellar 1.4.6 to 1.4.10 (Linux 64bit). Still no luck.

**Comment by rrahn** [comment]
Ok. We have to look at the tool pipeline because the maintainer of these apps are not available anymore. So just to make sure I understand you correctly. You want to do SV detection using Gustaf but you have to run Stellar before? 
Can you give us more specifics of the size of the reference genome, the number of reads and their length etc. The error typically indicates an allocation of to much main memory. That can be a bug or just something about the size of the data. It would be even best if you can give us the particular data set, but we can try first with simulated data.

**Comment by zijuexiansheng** [comment]
The reference genome

* total length: 4641702
* only 1 contig

reads before mate joining (Illumina Miseq read pairs)

* total length: 6702992347 + 6703466407
* number of read pairs: 22391084
* Picard statistics
    * mean insert size: 597.043745
    * standard deviation: 150.504076
    * pair orientation: FR

reads after mate joining

* total length: 13406458754
* number of reads: 22391084

**Comment by zijuexiansheng** [comment]
I added some code and noticed that Stellar threw the bad_alloc exception when allocating 34359738369 * 8 bytes (about 256 GB) of data. I was able to replay this exception for a few times using a simpler code, but after that, the exception was gone. But obviously, Stellar has more frequent allocate/deallocate behavior, so every time I run it, the exception will occur.

If there is a simpler way to record the allocate/deallocate behavior, I'd love to see how much memory has been allocated so that the allocation failed. But any way, it doesn't look like a bug to me.
